samza-core/src/main/java/org/apache/samza/execution/ExecutionPlanner.java

```diff
@@ -57,7 +57,7 @@ public ExecutionPlanner(Config config, StreamManager streamManager) {
     this.streamManager = streamManager;
   }
 
-  public JobGraph plan(StreamGraph streamGraph) throws Exception {
```

Comment:
Since StreamGraph is an user-facing API, IMHO it shouldn't expose getInputStreams/getOutputStreams (because users don't need them). Feels cleaner to move internal methods to the Impl class and use it directly. This is similar to how StreamOperatorTask uses MessageStreamImpl directly, how MessageStreamImpl uses StreamGraphImpl directly, etc. @